### PR TITLE
telemetry(amazonq): Adding cwsprChatProgrammingLanguage parameter to insertAtCursor and CopyAtClipboard events

### PR DESCRIPTION
**Pull request by laileni-aws**
## Problem
- Client side telemetry event `interactWithMessage` is missing `cwsprChatProgrammingLanguage` parameter which helps to understand the language of generated code.

## Solution
- Added `cwsprChatProgrammingLanguage` parameter for insertAtCursor and CopyAtClipboard events.
- No Test cases were added in this PR.
- Added `extractCodeBlockLanguage` function as current UTG Sync API sends python and java in the first chunk of generated code but thats the not case for other languages. To counter this, I am checking entire response for generated code language and emitting those metrics.
- Added `cwsprChatProgrammingLanguage` in [commons](https://github.com/aws/aws-toolkit-common) package: https://github.com/aws/aws-toolkit-common/pull/883
- Added `cwsprChatHasProjectContext and cwsprChatTotalCodeBlocks` in  in [commons](https://github.com/aws/aws-toolkit-common) package: https://github.com/aws/aws-toolkit-common/pull/884
---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts [hunk 1]
```diff
@@ -334,6 +342,24 @@ export class Messenger {
             })
     }
 
+    private extractCodeBlockLanguage(message: string): string {
```
Comment (justinmk3) [inline packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts:345]: seems like something that should be tested. and maybe Q team wants to define a `shared/markdown.ts` file (i don't know how much other things like this are in this or similar modules)

**Review by justinmk3 — APPROVED** [review]

File: packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts (hunk 1 above)
Comment (laileni-aws) [reply packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts]: Addressed this

File: packages/core/src/amazonq/webview/ui/main.ts [hunk 2]
```diff
@@ -582,7 +587,8 @@ export const createMynahUI = (
                 eventId,
                 codeBlockIndex,
                 totalCodeBlocks,
-                messageUserIntentMap.get(messageId) ?? undefined
+                responseMetadata.get(messageId)?.[0] ?? undefined,
```
Comment (ivikash) [inline packages/core/src/amazonq/webview/ui/main.ts:590]: We should add these to `ViewDiff` and `AcceptDiff` as well.

Comment (laileni-aws) [reply packages/core/src/amazonq/webview/ui/main.ts:590]: Yes, that can be in different PR as these two are unavailable for `Generate Tests(Beta)`.
@ivikash Correct me on this!

**Comment by justinmk3** [comment]
> npm ERR! Invalid: lock file's @aws-toolkits/telemetry@1.0.267 does not satisfy @aws-toolkits/telemetry@1.0.271


use `npm install --save-dev` when updating dependencies.

**Comment by justinmk3** [comment]
Thanks for upstreaming the metric to common. The build is failing because the old local metric still exists and overrides the common one. You need to delete it (in this PR): https://github.com/aws/aws-toolkit-vscode/blob/3b6197b2cfa8345a0775096f06e0983aec50102b/packages/core/src/shared/telemetry/vscodeTelemetry.json#L840

**Pull request merged**